runtime/lua/vim/inspector.lua

```diff
@@ -0,0 +1,207 @@
+local M = {}
+
+---@class InspectorFilter
+---@field syntax boolean include syntax based highlight groups (defaults to true)
+---@field treesitter boolean include treesitter based highlight groups (defaults to true)
+---@field extmarks boolean include extmarks (defaults to true)
+---@field extmarks_with_hl_group boolean when false, also include extmarks without a hl_group (defaults to true)
+---@field semantic_tokens boolean include semantic tokens (defaults to true)
+local defaults = {
+  syntax = true,
+  treesitter = true,
+  extmarks = true,
+  extmarks_with_hl_group = true,
+  semantic_tokens = true,
+}
+
+---Get all the items at a given buffer position
+---@param bufnr? buffer defaults to the current buffer
+---@param row? number 0-indexed row. Defaults to the current cursor
+---@param col? number 0-indexed col. Defaults to the current cursor
+---@param filter? InspectorFilter (table|nil) a table with key-value pairs to filter the items
+---               - syntax (boolean): include syntax based highlight groups (defaults to true)
+---               - treesitter (boolean): include treesitter based highlight groups (defaults to true)
+---               - extmarks (boolean): include extmarks (defaults to true)
+---               - extmarks_with_hl_group (boolean): when false, also include extmarks without a hl_group (defaults to true)
+---               - semantic_tokens (boolean): include semantic tokens (defaults to true)
+function M.get(bufnr, row, col, filter)
```

Comment:
To leave the door open for a future `win` parameter, how about this signature: 
```suggestion
function M.get(row, col, filter, {buf?, win?})
```

Reply:
I would keep it as is. If they want to inspect a certain window, they can just get the current buffer from that window and pass it to the function. No need for a `inspect_win`

Reply:
A buffer can be displayed in multiple windows.

Reply:
I think the use case is for something like the treesitter playground where a plugin wants to get or show information related to a window it opened itself.

Reply:
@zeertzjq that doesn't matter. We only use the window if row/col are not set, so if they pass in a buffer with a correct row/col, it will all work as expected

Reply:
```lua
  local buf = vim.api.nvim_win_get_buf(win)
  local cursor = vim.api.nvim_win_get_cursor(win)
  vim.inspect_pos(buf, cursor[1] - 1, cursor[1])
```

Reply:
Oh, I didn't notice that Justin edited his comment.